ur_controllers/src/gpio_controller.cpp

```diff
@@ -398,9 +398,9 @@ bool GPIOController::setSpeedSlider(ur_msgs::srv::SetSpeedSliderFraction::Reques
     command_interfaces_[CommandInterfaces::TARGET_SPEED_FRACTION_CMD].set_value(
         static_cast<double>(req->speed_slider_fraction));
 
-    while (command_interfaces_[CommandInterfaces::TARGET_SPEED_FRACTION_ASYNC_SUCCESS].get_value() == ASYNC_WAITING) {
-      // Asynchronouse wait until the hardware interface has set the slider value
-      std::this_thread::sleep_for(std::chrono::milliseconds(50));
+    if(!waitForAsyncCommand([&](){return command_interfaces_[CommandInterfaces::IO_ASYNC_SUCCESS].get_value();}))
+    {
+      RCLCPP_WARN(get_node()->get_logger(),"Could not verify that io was set. (This might happen when using the mocked interface)");
```

Comment:
```suggestion
    if(!waitForAsyncCommand([&](){return command_interfaces_[CommandInterfaces::TARGET_SPEED_FRACTION_ASYNC_SUCCESS].get_value();}))
    {
      RCLCPP_WARN(get_node()->get_logger(),"Could not verify that target speed fraction was set. (This might happen when using the mocked interface)");
```